packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -688,6 +698,53 @@ abstract class BucketBase extends Resource implements IBucket {
     });
   }
 
+  /**
+   * Adds a bucket notification event destination.
+   * @param event The event to trigger the notification
+   * @param dest The notification destination (Lambda, SNS Topic or SQS Queue)
+   *
+   * @param filters S3 object key filter rules to determine which objects
+   * trigger this event. Each filter must include a `prefix` and/or `suffix`
+   * that will be matched against the s3 object key. Refer to the S3 Developer Guide
+   * for details about allowed filter rules.
+   *
+   * @see https://docs.aws.amazon.com/AmazonS3/latest/dev/NotificationHowTo.html#notification-how-to-filtering
+   *
+   * @example
+   *
+   *    bucket.addEventNotification(EventType.OnObjectCreated, myLambda, 'home/myusername/*')
+   *
+   * @see
+   * https://docs.aws.amazon.com/AmazonS3/latest/dev/NotificationHowTo.html
+   */
+  public addEventNotification(event: EventType, dest: IBucketNotificationDestination, ...filters: NotificationKeyFilter[]) {
```

Comment:
The declaration should also be added to the `IBucket` interface.

We also need a unit test in [`bucket.test.ts`](https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-s3/test/bucket.test.ts)

Reply:
Added the `addEventNotification` to IBucket, just not sure about the code coverage part.